UPGRADING.md

```diff
@@ -1,5 +1,19 @@
 Upgrading Grape
 ===============
+### Upgrading to >= 1.0.0
+Prior to this version the endpoint was presented with params in the form of a Hashie::Mash. From 1.0.0 upwards Grape presents the parameters to the endpoint as an
+ActiveSupport::HashWithIndifferentAccess.
+
+To restore the previous behaviour you can specify Hashie::Mash using build_with in the params block;
+
+```ruby
+params do
+  build_with Grape::Extensions::Hashie::Mash::ParamBuilder
+  optional :color, type: String, default: 'blue', values: ['blue', 'red', 'green']
+end
+```
+
+The various options are documented in [Grape::DSL::Parameters](lib/grape/dsl/parameters.rb)
```

Comment:
I couldn't find a clean way via composition as you're mixing into API, but Request creates the params when Endpoint is mounted in API so going that route I ended up passing the class to instantiate through Endpoint and onto Request. Which pretty convoluted given we have a mechanism for sharing parameters across API and Endpoint via the Settings mixin.